hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/Iterators.scala

```diff
@@ -266,13 +266,14 @@ class RecordMergingFileIterator(logFiles: List[HoodieLogFile],
   private def serialize(curRowRecord: InternalRow): GenericRecord =
     serializer.serialize(curRowRecord).asInstanceOf[GenericRecord]
 
-  private def merge(curRow: InternalRow, newRecord: HoodieRecord[_]): Option[InternalRow] = {
+  private def merge(curRow: InternalRow, newRecord: HoodieRecord[_]): scala.Option[InternalRow] = {
```

Comment:
Don't forget to call this `merge` method even if `updatedRecordOpt.isEmpty` is `true`.

Reply:
@beyond1920, I modified the logic in the function `hasNextInternal`. Please see if it works correctly.